### PR TITLE
oasis core v24.0

### PR DESCRIPTION
**Pull request by ptrus**
Vendor v24.0. This release contains only additions (no breaking removals from v23) so we remove v23.0 and use v24.0 now for Eden types.

### CODE REVIEW

File: coreapi/v23.0/staking/api/address.go [hunk 1]
```diff
@@ -1,7 +0,0 @@
-package api
-
-import (
-	original "github.com/oasisprotocol/oasis-core/go/staking/api"
-)
-
-type Address = original.Address
```
Comment (pro-wh) [inline coreapi/v23.0/staking/api/address.go:7]: hm github doesn't detect this as moved

**Review by pro-wh — APPROVED** [review]
thanks. if it compiles

File: coreapi/v23.0/staking/api/address.go (hunk 1 above)
Comment (pro-wh) [reply coreapi/v23.0/staking/api/address.go:7]: isn't it in though?

File: coreapi/v24.0/staking/api/address.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package api
+
+import (
+        original "github.com/oasisprotocol/oasis-core/go/staking/api"
+)
+
+type Address = original.Address
```
Comment (pro-wh) [inline coreapi/v24.0/staking/api/address.go:7]: here

File: coreapi/v23.0/staking/api/address.go (hunk 1 above)
Comment (ptrus) [reply coreapi/v23.0/staking/api/address.go:7]: Oh, ok even better :) I didn't double-check.

**Pull request merged**